docs/scaling-hive.md

```diff
@@ -0,0 +1,47 @@
+
+In AWS, Hive performs best on c instance types. M instances are fine, but c instances are better.
+
+In general, the faster your etcd storage, the faster Hive will perform. c5.4xlarge has twice the EBS bandwidth as c5.2xlarge, and so Hive performs a bit better on c5.4xlarge (although in our opinion, not so much faster that c5.4xlarge is worth the premium over c5.2xlarge). In AWS, we recommend using io1 with 1200 iops for your masters.
+
+If you need to scale out the number of clusters you need to manage, use multiple Hive clusters -- approximately 1 Hive per 1000 clusters managed. Or fewer if you prefer to keep the blast radius smaller.
+
+## Install Pods
+
+Hive 1.x request 800 Mib of memory for each install pod. If you use m5.xlarge workers, you can support about (15 Gib / 800 Mib) install pods per worker -- so about 16. If you need to support more concurrent installs, you can use more workers, and/or workers with more memory. Install pods use barely any CPU.
+
+## Blocking I/O
+
+hive-controllers (where the syncset controllers run) uses blocking i/o. We may move to scale-out or non-blocking i/o in the future, but for now you should be aware that communication (mostly syncsets) to managed clusters can block other things.
+
+## Threads
+
+Hive supports configuring the number of goroutines per controller by editing values in HiveConfig. See [Using Hive](using-hive.md) for documentation on this. In general, set the number of goroutines to roughly the number of vCPUs on your workers. For c5.2xlarge, we use 10 goroutines. If you run Hive on larger hardware, you need to bump up the goroutines. If you have only 10 goroutines but 100 vCPUs, 90 of the vCPUs will be mostly unused. If Hive manages clusters that are on slow networks or have connectivity issues, you may want to use more goroutines to work around Hive's use of blocking i/o. Don't use too many goroutines, or that will actually cause a slowdown (i.e. don't just set it to 1000 or something).
+
+## SyncSets
+
+Pushing configuation to managed clusters via SyncSets is the most CPU-intensive and network-intensive thing that Hive does. We scale test Hive by mostly looking at how syncsets perform, since post-install this is what Hive spends the majority of its time doing.
+
+The metric we judge syncset performance by is "applies per second". The following table lists some typical applies/sec rates for various hardware combinations we've tested.
+
+|masters|workers|goroutines|applies/sec|
```

Comment:
Probably should swap masters for control-plane given everything going on.

Reply:
done